mount/temp.go

```diff
@@ -58,6 +58,24 @@ func WithTempMount(ctx context.Context, mounts []Mount, f func(root string) erro
 			}
 		}
 	}()
+
+	// The volatile option of overlayfs doesn't allow to mount again using the
```

Comment:
I think this is probably the right place to strip the option. Need to ensure the input mount array is not overwritten though.

Reply:
@natefive Sorry for late reply.

> should I create another function which modifies the options calls before calling WithTempMount then update those other calls to call the new function?

For my original comment, yes. But recently, I checked the code about starting container. It's unlikely to change the mount since there is no entrypoint for us to update it before passing to shim. 

https://github.com/containerd/containerd/blob/b6abda73b806d45b34221b3bd50992dc82cd886f/container.go#L244-L267

I think we should introduce a label like https://github.com/containerd/containerd/pull/6528. 

> Maybe we don't need to persist the label in boltdb. We can pass the label by gRPC context and get mount with the option when we want to get the mount. 

cc @dmcgowan